rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileDownloadRequestChain.java

```diff
@@ -103,17 +105,20 @@ public Long call() throws IOException
     File file = new File(localFile);
     if (!file.exists()) {
       log.debug("Creating localfile : " + localFile);
-      String metadataFilePath = CacheUtil.getMetadataFilePath(remotePath, conf);
-      File mdFile = new File(metadataFilePath);
-      if (mdFile.exists() && mdFile.length() > 0) {
-        // Making sure when a new file gets created, we invalidate the existing metadata file
-        bookKeeper.invalidateFileMetadata(remotePath);
-      }
       file.setWritable(true, false);
       file.setReadable(true, false);
       file.createNewFile();
     }
 
+    long highestReadRequestLength = readRequests
+            .stream()
+            .map(readRequest -> readRequest.getActualReadEnd() - readRequest.getActualReadStart())
```

Comment:
Can use `readRequest.getActualReadLength`

Reply:
that can cause int overflow as `readRequest.getActualReadEnd() - readRequest.getActualReadStart()` can be huge in FileDownloadRRC

Reply:
Can we remove the cast from that method then ? It seems depending on context that method may or may not be safe to use which can be lead to mistakes in future.

Reply:
I have change the methods returning ints to call out they are Unsafe.